extras/dynflow@.service

```
@@ -0,0 +1,13 @@
+# /usr/lib/systemd/system/dynflow@.service
+[Unit]
+Description=Dynflow sidekiq worker
+After=network.target remote-fs.target nss-lookup.target
+
+[Service]
+TimeoutSec=90
+ExecStart=/usr/bin/sh -c 'exec sidekiq -C /etc/dynflow/%i.yml $DYNFLOW_OPTIONS'
```

```suggestion
ExecStart=/usr/bin/sh -c 'bundle exec sidekiq -C /etc/dynflow/%i.yml $DYNFLOW_OPTIONS'
```
Or
```suggestion
ExecStart=/usr/bin/bundle exec sidekiq -C /etc/dynflow/%i.yml $DYNFLOW_OPTIONS
```
Or (and change env file accordingly)
```suggestion
ExecStart=/usr/bin/bundle exec sidekiq -e $RAILS_ENV -r PATH_TO_SCRIPT -C /etc/dynflow/%i.yml 
```

---

The shell was there to make the env variables work. As far as I know systemd doesn't translate those, so that rules out numbers 2 and 3.

The exec was there to replace the shell process with the sidekiq one so we don't keep yet another process around. Also I'm not sure how good idea is to use `bundle` on systems where sidekiq is installed from rpms and uses system ruby.

---

Me neighter, so let's not. Thanks for clarifying.

---

Oh, so according to the docs[1, seventh paragraph] systemd can do env var substitution, just the behavior is slightly different from shell. That would mean I could drop the shell part

[1] - https://www.freedesktop.org/software/systemd/man/systemd.service.html#Command%20lines